### DISCUSSION TITLE
Usb Host Mouse for F1c100s

### DISCUSSION THREAD

**Discussion by satar1980** [hathach/tinyusb#1707] [Ideas]
I need example Usb Host Mouse for F1c100s, can you help me please..........

**Comment by manuelbl** [comment]
The helpful examples are:

- [host/hid_controller](https://github.com/hathach/tinyusb/tree/master/examples/host/hid_controller): It provides the basic structure for a USB HID host (but is built for a game controller instead of a mouse)
- [dual/host_hid_to_device_cdc](https://github.com/hathach/tinyusb/blob/master/examples/dual/host_hid_to_device_cdc/src/main.c): It works as both host and device but it contains the relevant code for mouse handling (see [main.c, line 271](https://github.com/hathach/tinyusb/blob/master/examples/dual/host_hid_to_device_cdc/src/main.c#L271)).

**Comment by satar1980** [comment]
I try to compile  host_hid_to_device_cdc and found this error : 

LINK _build/f1c100s/host_hid_to_device_cdc.elf
_build/f1c100s/obj/src/host/usbh.o: In function `process_device_unplugged':
/opt/f1c200/tinyusb/src/host/usbh.c:1144: undefined reference to `hcd_device_close'
_build/f1c100s/obj/src/host/usbh.o: In function `usbh_int_set':
/opt/f1c200/tinyusb/src/host/usbh.c:727: undefined reference to `hcd_int_enable'
/opt/f1c200/tinyusb/src/host/usbh.c:730: undefined reference to `hcd_int_disable'
_build/f1c100s/obj/src/host/usbh.o: In function `osal_task_delay':
/opt/f1c200/tinyusb/src/host/usbh.c:282: undefined reference to `hcd_frame_number'
/opt/f1c200/tinyusb/src/host/usbh.c:283: undefined reference to `hcd_frame_number'
_build/f1c100s/obj/src/host/usbh.o: In function `tuh_init':
/opt/f1c200/tinyusb/src/host/usbh.c:380: undefined reference to `hcd_init'
/opt/f1c200/tinyusb/src/host/usbh.c:381: undefined reference to `hcd_int_enable'
_build/f1c100s/obj/src/host/usbh.o: In function `usbh_edpt_control_open':
/opt/f1c200/tinyusb/src/host/usbh.c:823: undefined reference to `hcd_edpt_open'
_build/f1c100s/obj/src/host/usbh.o: In function `usbh_control_xfer_cb':
/opt/f1c200/tinyusb/src/host/usbh.c:671: undefined reference to `hcd_edpt_xfer'
_build/f1c100s/obj/src/host/usbh.o: In function `usbh_edpt_xfer_with_callback':
/opt/f1c200/tinyusb/src/host/usbh.c:794: undefined reference to `hcd_edpt_xfer'
_build/f1c100s/obj/src/host/usbh.o: In function `tuh_edpt_open':
/opt/f1c200/tinyusb/src/host/usbh.c:830: undefined reference to `hcd_edpt_open'
_build/f1c100s/obj/src/host/usbh.o: In function `process_enumeration':
/opt/f1c200/tinyusb/src/host/usbh.c:1292: undefined reference to `hcd_device_close'
_build/f1c100s/obj/src/host/usbh.o: In function `enum_new_device':
/opt/f1c200/tinyusb/src/host/usbh.c:1382: undefined reference to `hcd_port_reset'
/opt/f1c200/tinyusb/src/host/usbh.c:1385: undefined reference to `hcd_port_reset_end'
/opt/f1c200/tinyusb/src/host/usbh.c:1388: undefined reference to `hcd_port_connect_status'
/opt/f1c200/tinyusb/src/host/usbh.c:1390: undefined reference to `hcd_port_speed_get'
_build/f1c100s/obj/src/host/usbh.o: In function `tuh_control_xfer':
/opt/f1c200/tinyusb/src/host/usbh.c:567: undefined reference to `hcd_setup_send'
/opt/f1c200/tinyusb/src/host/usbh.c:578: undefined reference to `hcd_setup_send'
collect2: error: ld returned 1 exit status
../../rules.mk:100: recipe for target '_build/f1c100s/host_hid_to_device_cdc.elf' failed

how to fix it

**Reply by manuelbl** [reply]
It seems I have overlooked that TinyUSB's implementation for f1c100s only supports device mode. I'm not entirely sure as the documentation is incomplete.
